pkg/apis/extensions/types.go

```diff
@@ -506,8 +505,20 @@ type DaemonSet struct {
 	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
 	// +optional
 	Status DaemonSetStatus
+
+	// A sequence number representing a specific generation of the template.
+	// Populated by the system. Read-only.
+	// +optional
+	TemplateGeneration int64
```

Comment:
Let's move it to spec. We also need to update its comment to say that we allow users to set this value when creating a DaemonSet, see [proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/daemonset-update.md)